doc/modules/preprocessing.rst

```diff
@@ -424,5 +425,8 @@ Note that, here, missing values are encoded by 0 and are thus implicitly stored
 in the matrix. This format is thus suitable when there are many more missing
 values than observed values.
 
+Also, knn imputation strategy will use samples with full features, and if all samples
```

Comment:
I'd remove the filler "also" and say "When using ``strategy='knn'``, only samples without any missing features will be used for imputation. If all samples have missing features, this strategy will fail.".